contracts/modules/cover/Cover.sol

```diff
@@ -0,0 +1,588 @@
+    Cover memory cover = covers[coverId];
+
+    StakingPool[] storage currentStakingPools = stakingPoolsForCover[covers.length];
+
+    uint32 period = uint32(block.timestamp) - cover.start;
+    // convert to NXM amount
+    uint amountToCover = uint(amount) * 1e18 / pool().getTokenPrice(pool().assets(cover.payoutAsset));
+    uint totalPremiumInNXM = 0;
+    for (uint i = 0; i < stakingPools.length; i++) {
+      if (amountToCover == 0) {
+        break;
+      }
+
+      IStakingPool stakingPool = IStakingPool(stakingPools[i].poolAddress);
+      (uint coveredAmount, uint premiumInNXM) = buyCoverFromPool(stakingPool, cover.productId, amountToCover, period);
+      amountToCover -= coveredAmount;
+      totalPremiumInNXM += premiumInNXM;
+
+      uint j = 0;
+      for ( ; j < currentStakingPools.length; j++) {
+        if (currentStakingPools[j].poolAddress == stakingPools[i].poolAddress) {
+          currentStakingPools[j].coverAmount += uint96(coveredAmount);
+          currentStakingPools[j].premiumInNXM += uint96(premiumInNXM);
+          break;
+        }
+      }
+
+      if (j < currentStakingPools.length) {
+        continue;
+      }
+
+      stakingPoolsForCover[covers.length].push(
+        StakingPool(
+          address(stakingPool),
+          uint96(coveredAmount),
+          uint96(premiumInNXM)
+        ));
+    }
+    require(amountToCover == 0, "Not enough available capacity");
+
+    premiumInAsset = totalPremiumInNXM * pool().getTokenPrice(pool().assets(cover.payoutAsset)) / 1e18;
+
+    // make the previous cover expire at current block
+    uint32 newPeriod = uint32(block.timestamp) - cover.start;
+    uint32 previousPeriod = covers[coverId].period;
+
+    cover.amount += amount;
+    cover.premium += uint96(premiumInAsset);
+    cover.start = uint32(block.timestamp);
+    cover.period = uint32(block.timestamp) - previousCover.start;
+
+    covers.push(cover);
+
+    newCoverId = covers.length - 1;
+
+    // mint the new cover
+    _safeMint(msg.sender, newCoverId);
+  }
+
+  function addPeriod(uint coverId, uint32 extraPeriod, uint maxPrice) external payable {
+
+    require(msg.sender == ERC721.ownerOf(coverId), "Cover: not cover owner");
+    uint premiumInAsset = _addPeriod(coverId, extraPeriod);
+    require(premiumInAsset <= maxPrice, "Cover: Price exceeds maxPrice");
+
+    retrievePayment(premiumInAsset, covers[coverId].payoutAsset);
+  }
+
+  function _addPeriod(uint coverId, uint32 extraPeriod) internal returns (uint) {
+
+    Cover storage cover = covers[coverId];
+    StakingPool[] storage stakingPools = stakingPoolsForCover[covers.length];
+
+    uint totalPremiumInNXM = 0;
+    for (uint i = 0; i < stakingPools.length; i++) {
+      IStakingPool stakingPool = IStakingPool(stakingPools[i].poolAddress);
+
+      uint capacityFactor = capacityFactors[cover.productId];
+      (uint basePrice, uint premiumInNXM) = getPrice(cover.productId, extraPeriod, cover.productId, stakingPool);
```

Comment:
you should exclude the cover amount from the used capacity

Reply:
you're passing the `productId` instead of the `amount` to the `getPrice` function